test/decl/protocol/req/associated_type_inference.swift

```diff
@@ -563,51 +564,6 @@ extension S31 where T: P32 {
   func boo() -> Void {}
 }
 
-// SR-12707
-
-class SR_12707_C<T> {}
-
-// Inference in the adoptee
-protocol SR_12707_P1 {
-  associatedtype A
-  associatedtype B: SR_12707_C<(A, Self)>
```

Comment:
Moved these SR tests down so that I could switch to the established numeric naming scheme and keep the order.